sw/airborne/firmwares/rotorcraft/navigation.c

```diff
@@ -74,8 +74,12 @@ PRINT_CONFIG_VAR(NAVIGATION_FREQUENCY)
 #define FAILSAFE_MODE_DISTANCE (1.5*MAX_DIST_FROM_HOME)
 #endif
 
+#ifndef NAV_CARROT_DIST
+#define NAV_CARROT_DIST 12
+#endif
+
 #define CLOSE_TO_WAYPOINT (15 << INT32_POS_FRAC)
-#define CARROT_DIST (12 << INT32_POS_FRAC)
+#define CARROT_DIST ( (int32_t) NAV_CARROT_DIST << INT32_POS_FRAC)
```

Comment:
with this, you "loose" the sub-metric precision (NAV_CARROT_DIST is cast to int before shifting). I suggest:
```suggestion
#define CARROT_DIST ((int32_t) POS_BFP_OF_REAL(NAV_CARROT_DIST))
```

Reply:
Ah yes of course, good one!